presto-hive/src/main/java/com/facebook/presto/hive/metastore/thrift/Transport.java

```diff
@@ -47,44 +54,37 @@ public static TTransport create(HostAndPort address, Optional<HostAndPort> socks
 
     private Transport() {}
 
-    private static TTransport createRaw(HostAndPort address, Optional<HostAndPort> socksProxy, int timeoutMillis)
+    private static TTransport createRaw(HostAndPort address, Optional<SSLContext> sslContext, Optional<HostAndPort> socksProxy, int timeoutMillis)
             throws TTransportException
     {
-        if (!socksProxy.isPresent()) {
-            return new TSocket(address.getHost(), address.getPort(), timeoutMillis);
-        }
+        Proxy proxy = socksProxy
+                .map(socksAddress -> new Proxy(SOCKS, InetSocketAddress.createUnresolved(socksAddress.getHost(), socksAddress.getPort())))
+                .orElse(Proxy.NO_PROXY);
 
-        Socket socks = createSocksSocket(socksProxy.get());
+        Socket socket = new Socket(proxy);
         try {
+            // setSocketProperties(socket);
+            socket.connect(new InetSocketAddress(address.getHost(), address.getPort()), timeoutMillis);
+
+            if (sslContext.isPresent()) {
+                // SSL connect is to the socks address when present
+                HostAndPort sslConnectAddress = socksProxy.orElse(address);
+
+                socket = sslContext.get().getSocketFactory().createSocket(socket, sslConnectAddress.getHost(), sslConnectAddress.getPort(), true);
+                // setSocketProperties(socket);
+            }
+            return new TSocket(socket);
+        }
+        catch (Throwable t) {
+            // something went wrong, close the socket and rethrow
             try {
-                socks.connect(InetSocketAddress.createUnresolved(address.getHost(), address.getPort()), timeoutMillis);
-                socks.setSoTimeout(timeoutMillis);
-                return new TSocket(socks);
+                socket.close();
             }
-            catch (Throwable t) {
-                closeQuietly(socks);
-                throw t;
+            catch (IOException e) {
+                t.addSuppressed(e);
```

Comment:
Looks like promoting `io.airlift.testing.Closeables#closeAllSuppress` to non-testing would be beneficial here (and not only here).